[OC-120] Registry webUI: preview route/view bug

nickbalestra opened the pull request:
Preview route didnt take into account of registry prefix for paths

matteofigus commented:
The baseUrl should be set to contain the prefix too. Perhaps we should improve the baseUrl docs?

nickbalestra commented:
Mmmm I see...
Yes totally not clear.
My personal feeling is that if you use two seperate configuration then the two things should be indipendent, no?

nickbalestra commented:
I have an idea 😎 , let me rework this PR

nickbalestra commented:
@matteofigus I've extended the `options-sanitizer`:
It now checks if a prefix exist (before sanitizing it) and if the baseUrl doesn't already contain the prefix. If both check succeed it it sanitize the baseUrl by appending the prefix to it.

This should avoid this being a braking change, and will make it possible to handle both cases:
1. prefix and baseUrl containing prefix
2. prefix and baseUrl without prefix

Let me know what you think

nickbalestra commented:
add
- [ ] sanitize baseUrl with trailing slash as per https://github.com/opencomponents/get-started/issues/2

nickbalestra commented:
@matteofigus do want to review this one more time before 🌴 🌞  🏄  🍺  🏖  ?

nickbalestra commented:
@mattiaerre to get into merge mood, what about starting with this? 😎

nickbalestra commented:
There are still a couple of issues that need to be addressed that seems the tests didn't cought:
if prefix && baseUrl contiene prefix, niente duplicate prefix...
I'll need to add more tests to cover also this scenario

nickbalestra commented:
@mattiaerre I've rebased and added the scenario based tests for the the baseUrl/Prefix.

I'll rather keep this PR scoped on solving this very small 'bug' that we discovered with the heroku get started registry.
I therefore suggest we don't addd jest or takle other aspects of the sanitization like `beforePublish` yet. (Also no need to add jest to have snapshots as those are now avail for any test runner).

Can you open an issue for the `beforePublish`? Ffor jest we already have #473 
If Ok i'll revert the commit that added jest.

nickbalestra commented:
@mattiaerre this should be now ready to go.
I'll update the OpenComponents website registry and the get-started repo once this is merged. Thanks!

mattiaerre commented on .gitignore:
```diff
@@ -27,4 +27,4 @@ src/components/base-component-handlebars/_package
 test/fixtures/test.tar.gz
-test/fixtures/targz-test
\ No newline at end of file
+test/fixtures/targz-test
```
me love this!

mattiaerre commented on test/unit/registry-domain-options-sanitiser.js:
```diff
@@ -69,4 +76,48 @@ describe('registry : domain : options-sanitiser', () => {
   });
+
+  describe('prefix and baseUrl sanitaziation', () => {
+    const prefixAndBaseUrlSchenario = [
```
love it!

mattiaerre replied:
typo `prefixAndBaseUrlSchenario` => `prefixAndBaseUrlScenarios`

mattiaerre commented on test/unit/registry-domain-options-sanitiser.js:
```diff
@@ -7,7 +7,7 @@ describe('registry : domain : options-sanitiser', () => {
   const sanitise = require('../../src/registry/domain/options-sanitiser');
 
   describe('when options is empty', () => {
```
this is no longer the description for this test is it?

mattiaerre commented on test/unit/registry-domain-options-sanitiser.js:
```diff
@@ -69,4 +76,48 @@ describe('registry : domain : options-sanitiser', () => {
     });
   });
+
+  describe('prefix and baseUrl sanitaziation', () => {
```
typo `sanitaziation` => `sanitization`
